frontend_vue/src/components/ui/IncidentDetails.vue

```diff
@@ -1,80 +1,145 @@
-            Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36
-            (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36
-          </p>
-        </li>
-        <li>
-          <span class="block text-xs uppercase text-grey-500"
-            >request headers</span
-          >
-          <code class="text-grey-800">
-            {'Host': 'canarytokens.com', 'X-Real-Ip': '84.138.198.174',
-            'X-Forwarded-For': '84.138.198.174', 'X-Forwarded-Host':
-            'canarytokens.org', 'Connection': 'close',
-            'Upgrade-Insecure-Requests': '1', 'User-Agent': 'Mozilla/5.0
-            (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like
-            Gecko) Chrome/122.0.0.0 Safari/537.36', 'Accept':
-            'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7',
-            'Accept-Encoding': 'gzip, deflate', 'Accept-Language':
-            'en-GB,en-US;q=0.9,en;q=0.8,zh-CN;q=0.7,zh;q=0.6,it;q=0.5'}
-          </code>
-        </li>
-      </ul>
-    </section>
+            <template v-if="isNotEmptyObject(val)">
+              <li
+                v-for="(subval, subkey) in val"
+                :key="subkey"
+                class="break-words"
+              >
+                <span class="block text-xs uppercase text-grey-500">{{
+                  subkey
+                }}</span>
+                <span :class="addClassToBoolean(val)">{{ subval }}</span>
+              </li>
+            </template>
+            <template v-else>
+              <li
+                :key="key"
+                :class="addClassToBoolean(val)"
+                class="break-words"
+              >
+                {{ val }}
+              </li>
+            </template>
+          </ul>
+        </template>
+      </section>
+    </div>
   </div>
 </template>
 
 <script setup lang="ts">
-const emit = defineEmits<{
-  (e: 'update:modelValue', value: boolean): void;
+import { computed } from 'vue';
+import type { HitsType } from '@/components/tokens/types.ts';
+import { convertUnixTimeStampToDate } from '@/utils/utils';
+
+const props = defineProps<{
+  hitAlert: HitsType;
 }>();
+
+const emit = defineEmits(['close']);
+
+console.log(props.hitAlert, 'hitsList');
+
+function isNotEmptyObject(
+  val: boolean | string[] | undefined | HitsType
+): boolean {
+  return typeof val === 'object' && Object.keys(val)?.length > 0;
+}
+
+function addClassToBoolean(val: boolean | HitsType | undefined) {
+  if (typeof val === 'boolean') {
+    return val ? 'text-green' : 'text-red';
+  }
+}
+
+function cleanupObject(obj: keyof HitsType) {
+  return Object.entries(obj).reduce((acc, [key, value]) => {
+    if (value !== null && value !== undefined) {
+      //@ts-ignore
+      acc[key as keyof HitsType] =
+        typeof value === 'object' ? cleanupObject(value) : value;
+    }
+    console.log(acc);
+    return acc;
+  }, {} as HitsType);
+}
```

Comment:
what are we doing here, just stripping out null values?

Reply:
are we sure we won't need those somewhere?

Reply:
uhm I'm not sure about it, do you have a scenario in mind, where a null value is important for the user? 🧐
Thing is we get a lot of empty obj/null values from the backend. I'm not sure we want to show all of them to the user

Reply:
so i'm going to spin this back to you and as are you sure we don't display empty things anywhere in the triggered alert list?